pkgs/development/libraries/openssl/default.nix

```diff
@@ -58,6 +59,9 @@ stdenv.mkDerivation rec {
 
     # remove dependency on Perl at runtime
     rm -r $out/etc/ssl/misc $out/bin/c_rehash
+
+    # configure the default trust store
+    ${optionalString (defaultCertificate != null) "ln -s ${defaultCertificate} $out/etc/ssl/cert.pem"}
```

Comment:
We could override X509_CERT_FILE, but it creates more impurities at build time IIRC.
In this case, we add something at the end of postInstall, this cannot harm much.

I made it configurable because it may not be desirable to have a reference to "/etc/ssl/certs/ca-certificates.crt" on darwin for example. I am not too sure about this.